vendor/processors/stripe_processor.py

```diff
@@ -656,12 +676,19 @@ def update_offers(self, offers):
 
     def sync_offers(self, site):
         products = self.get_site_offers(site)
-        offer_pk_list = [product['metadata']['pk'] for product in products]
+        offer_pk_list = []
+
+        if products:
+            offer_pk_list = [product['metadata']['pk'] for product in products]
+
         offers_in_vendor = self.get_vendor_offers_in_stripe(offer_pk_list, site)
         offers_not_in_vendor = self.get_vendor_offers_not_in_stripe(offer_pk_list, site)
 
-        self.create_offers(offers_not_in_vendor)
-        self.update_offers(offers_in_vendor)
+        if offers_not_in_vendor:
+            self.create_offers(offers_not_in_vendor)
+
+        if offers_in_vendor:
```

Comment:
`if` condition is not neccessary. Remove it and let it call the function with an empty list